src/plugins/intel_gna/legacy/src/convert_function_to_cnn_network.cpp

```diff
@@ -1968,15 +1970,26 @@ CNNLayerCreator::CNNLayerCreator(const std::shared_ptr<::ngraph::Node>& node) :
                        });
 }
 
-CNNLayerPtr CNNLayerCreator::create() {
-    LayerParams attrs = {node->get_friendly_name(),
-                         node->description(),
-                         details::convertPrecision(node->get_output_element_type(0))};
-    if (creators.find(node->description()) != creators.end())
-        return creators[node->description()](node, params);
+CNNLayerPtr CNNLayerCreator::create(const std::shared_ptr<::ngraph::Node>& node_) {
+    node = node_;  // node used by node->visit_attributes(..) > AttributeVisitor::on_attribute(..)
+
+    if (!node_->visit_attributes(*this))
+        return nullptr;
+
+    LayerParams attrs = {node_->get_friendly_name(),
+                         node_->description(),
+                         details::convertPrecision(node_->get_output_element_type(0))};
+
+    CNNLayerPtr res;
+
+    if (creators.find(node_->description()) != creators.end())
```

Comment:
I guess one search loop instead of two would be faster if we want to improve performance.

```suggestion
    auto creator_iterator = creators.find(node_->description());
    if ( creator_iterator != creators.end()){
        res = *creator_iterator(node_, params);
    }
```

Reply:
indeed. changed.

Reply:
thank you for all comments